tqdm/std.py

```diff
@@ -1307,9 +1335,7 @@ def close(self):
         # Prevent multiple closures
         self.disable = True
 
-        # decrement instance pos and remove from internal set
         pos = abs(self.pos)
-        self._decr_instances(self)
```

Comment:
Have to combine the locking inside of `_decr_instances` and the locking located some lines below to avoid a race condition, which may result in 2 (or more) bars terminating lines, while only the very last one must